pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/TableUpsertMetadataManager.java

```diff
@@ -50,4 +53,6 @@ public interface TableUpsertMetadataManager extends Closeable {
    * @return A {@code Map} where keys are partition id and values are count of primary keys for that specific partition
    */
   Map<Integer, Long> getPartitionToPrimaryKeyCount();
+
+  List<SegmentContext> getSegmentContexts(List<IndexSegment> selectedSegments);
```

Comment:
Should we pass in `List<SegmentContext>` and return `List<SegmentContext>`? Currently upsert is the only component decorating segment, but we might add more in the future

Reply:
good point